runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkContextFactory.java

```diff
@@ -85,6 +86,8 @@ private static JavaSparkContext createSparkContext(SparkPipelineOptions options)
         conf.setMaster(options.getSparkMaster());
       }
       conf.setAppName(options.getAppName());
+      // register immutable collections serializers because the SDK uses them.
+      conf.set("spark.kryo.registrator", BeamSparkRunnerRegistrator.class.getCanonicalName());
```

Comment:
I'm assuming that Spark loads things by doing Class.forName. If thats the case then the canonical name is the incorrect one since it won't allow for loading static inner classes.

This same comment applies to spark.serializer below.

Given an inner class called Foo inside of Bar:
canonical name = "package.Foo.Bar"
name = "package.Foo$Bar"

Class.forName would work with the name but throw if given the canonical name.